### PR TITLE
feat: Surface parameters fields for database model

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Update the api to surface parameters in GET `database`. This will be leveraged for whenever a user wants to edit a database assuming configuration type is dynamic.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [X] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/database/types.ts [hunk 1]
```diff
@@ -21,6 +21,16 @@ type DatabaseUser = {
   last_name: string;
 };
 
+type DatabaseParameters = {
+  database: string;
+  engine: string;
+  host: string;
+  password?: string;
+  port: number;
+  query: Object;
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/database/types.ts:30]: I think it is better to use the query props here, which I think you can import.

Comment (betodealmeida) [reply superset-frontend/src/views/CRUD/data/database/types.ts:30]: Confusingly, this is a different query: https://docs.sqlalchemy.org/en/14/core/engines.html#add-parameters-to-the-url-query-string

Comment (AAfghahi) [reply superset-frontend/src/views/CRUD/data/database/types.ts:30]: oh interesting, so then would Record<string, string> be more explicit here given that it has string keys and values?

Comment (eschutho) [reply superset-frontend/src/views/CRUD/data/database/types.ts:30]: That sounds like a good plan.

Comment (eschutho) [reply superset-frontend/src/views/CRUD/data/database/types.ts:30]: I also added in some props in https://github.com/apache/superset/pull/14583. For bigquery most of those props are going be optional (maybe some other dbs too). Are we going to use db parameters for Gsheets for example?

File: superset/models/core.py [hunk 2]
```diff
@@ -222,6 +223,17 @@ def unique_name(self) -> str:
     def url_object(self) -> URL:
         return make_url(self.sqlalchemy_uri_decrypted)
 
+    @property
+    def parameters(self) -> Optional[Dict[str, Any]]:
+        # Build parameters if db_engine_spec is a subclass of BasicParametersMixin
+        parameters = {"engine": self.backend}
```
Comment (eschutho) [inline superset/models/core.py:229]: so for sqlalchemy forms, we're only going to return engine? Maybe that should be the only required type then, it looks like.

File: superset-frontend/src/views/CRUD/data/database/types.ts (hunk 1 above)
Comment (betodealmeida) [reply superset-frontend/src/views/CRUD/data/database/types.ts:30]: @AAfghahi right, `Record<string, string>`

But we need to keep in mind that these are specific for Postgres and similar databases. For other DBs some attributes might be optional, or there might be a different set of attributes, so maybe we should just define the type as `Object`.

File: superset/models/core.py [hunk 3]
```diff
@@ -222,6 +223,17 @@ def unique_name(self) -> str:
...
+
+        if issubclass(self.db_engine_spec, BasicParametersMixin):
```
Comment (betodealmeida) [inline superset/models/core.py:231]: After we add BQ we need to change this, right?

File: superset-frontend/src/views/CRUD/data/database/types.ts (hunk 1 above)
Comment (AAfghahi) [reply superset-frontend/src/views/CRUD/data/database/types.ts:30]: that's fair, I hadn't considered that.

File: superset/models/core.py (hunk 2 above)
Comment (hughhhh) [reply superset/models/core.py:229]: Yea good catch i'll make sure to update the type on the types file

File: superset/models/core.py (hunk 3 above)
Comment (hughhhh) [reply superset/models/core.py:231]: Yea that PR will have the updated logic here to allow us to pass Big Query credentials

**Comment by hughhhh** [comment]
closing in favor of #14653

**Pull request closed without merging**